hack/jenkins/e2e.sh

```diff
@@ -236,7 +236,12 @@ fi
 # Jenkins will look at the junit*.xml files for test failures, so don't exit
 # with a nonzero error code if it was only tests that failed.
 if [[ "${E2E_TEST,,}" == "true" ]]; then
-    go run ./hack/e2e.go ${E2E_OPT} -v --test --test_args="${GINKGO_TEST_ARGS} --ginkgo.noColor" || true
+    go run ./hack/e2e.go ${E2E_OPT} -v --test --test_args="${GINKGO_TEST_ARGS} --ginkgo.noColor" && exitcode=0 || exitcode=$?
+    if [[ ${E2E_PUBLISH_GREEN_VERSION:-} == "true" && $exitcode == 0 && -n ${githash:-} ]]; then
```

Comment:
can't decide whether it's relevant if cleanup succeeded as well in marking a release as green.